vispy/util/tests/test_event_emitter.py

```diff
@@ -663,4 +663,81 @@ def assert_state(a, b):
     assert_state(True, True)
 
 
+def test_emitter_reentrance_allowed_when_blocked1():
+    # Minimal re-entrance example
+
+    e = EventEmitter(source=None, type='test')
+    count = 0
+
+    @e.connect
+    def foo(x):
+        nonlocal count
+        count += 1
+        with e.blocker():
+            e()
+
+    e()
+    assert count == 1
+
+
+def test_emitter_reentrance_allowed_when_blocked2():
+    # More realistic re-entrance example
+    # The real world case for this might be a longer chain of events
+    # where event1's callback knows that what it's doing will trigger
+    # the same event so it blocks it. So event1 -> foo -> event2
+    # -> bar -> event1 -> (ignored bc blocked).
```

Comment:
@djhoese I included your explanation because that helped me figure out the use-case :)